src/filesystem_stream.h

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef EP_FILESYSTEM_STREAM_H
+#define EP_FILESYSTEM_STREAM_H
+
+// Headers
+#include <cassert>
+#include <istream>
+#include <ostream>
+#include "utils.h"
+
+namespace Filesystem_Stream {
+	class InputStream : public std::istream {
+	public:
+		InputStream() = default;
+		explicit InputStream(std::streambuf* sb, std::streamsize size);
+		~InputStream() override;
+		InputStream(const InputStream&) = delete;
+		InputStream& operator=(const InputStream&) = delete;
+		InputStream(InputStream&& is) : std::istream(std::move(is)) {
+			set_rdbuf(is.rdbuf());
+			is.set_rdbuf(nullptr);
```

Comment:
You forgot to move `streamsize`. Since we're setting rdbuf to nullptr in `is`, we should also make `is.streamsize = 0`